cmake/modules/FindoneApi.cmake

```
@@ -0,0 +1,30 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2023 Intel Corporation
+#
+# FindoneApi module for locating oneAPI compiler, icx.
+#
+# The module defines the following variables:
+#
+# 'oneApi_FOUND', 'ONEAPI_FOUND'
+# True if the oneApi toolchain/compiler, icx, was found.
+#
+# 'ONEAPI_VERSION'
+# The version of the oneAPI toolchain.
+
+if(CMAKE_C_COMPILER)
+  # Parse the 'clang --version' output to find the installed version.
+  execute_process(COMMAND ${CMAKE_C_COMPILER} --version OUTPUT_VARIABLE ONEAPI_VERSION)
+  string(REGEX REPLACE "[^0-9]*([0-9.]+) .*" "\\1" ONEAPI_VERSION ${ONEAPI_VERSION})
+endif()
+
+find_package_handle_standard_args(oneApi
+				  REQUIRED_VARS CMAKE_C_COMPILER
+				  VERSION_VAR ONEAPI_VERSION
+)
+
+if(NOT OneApi_FOUND)
+  # oneApi was found but version requirement is not met.
+  # Treat it as CMAKE_C_COMPILER was never found by resetting the result from `find_program()`
+  set(CMAKE_C_COMPILER ONEAPI_NOTFOUND CACHE FILEPATH "Path to a program" FORCE)
```

did you mean: `CMAKE_C_COMPILER-NOTFOUND` as per:
Ref: https://cmake.org/cmake/help/latest/command/find_program.html
>  If nothing is found, the result will be <VAR>-NOTFOUND.

```suggestion
  set(CMAKE_C_COMPILER CMAKE_C_COMPILER-NOTFOUND CACHE FILEPATH "Path to a program" FORCE)
```

---

Hmm, if I do this the `cmake -DBOARD=qemu_x86` does not fail/error.  It reports the incorrect version, but happily succeeds.  Seems like we should fail at this stage some how.

---

If you add `REQUIRED` as described here: https://github.com/zephyrproject-rtos/zephyr/pull/54075#discussion_r1115551473

then this code can probably be removed.